src/Processors/Formats/Impl/NpyRowInputFormat.cpp

```diff
@@ -0,0 +1,434 @@
+        column_int16->insertValue(value);
+    }
+    else if (auto * column_int32 = typeid_cast<ColumnInt32 *>(column.get()))
+    {
+        Int32 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_int32->insertValue(value);
+    }
+    else if (auto * column_int64 = typeid_cast<ColumnInt64 *>(column.get()))
+    {
+        Int64 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_int64->insertValue(value);
+    }
+    else if (auto * column_uint8 = typeid_cast<ColumnUInt8 *>(column.get()))
+    {
+        UInt8 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_uint8->insertValue(value);
+    }
+    else if (auto * column_uint16 = typeid_cast<ColumnUInt16 *>(column.get()))
+    {
+        UInt16 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_uint16->insertValue(value);
+    }
+    else if (auto * column_uint32 = typeid_cast<ColumnUInt32 *>(column.get()))
+    {
+        UInt32 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_uint32->insertValue(value);
+    }
+    else if (auto * column_uint64 = typeid_cast<ColumnUInt64 *>(column.get()))
+    {
+        UInt64 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_uint64->insertValue(value);
+    }
+    else if (auto * column_float32 = typeid_cast<ColumnFloat32 *>(column.get()))
+    {
+        Float32 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_float32->insertValue(value);
+    }
+    else if (auto * column_float64 = typeid_cast<ColumnFloat64 *>(column.get()))
+    {
+        Float64 value = 0;
+        endian == 1 ? readBinaryBigEndian(value, *in) : readBinaryLittleEndian(value, *in);
+        column_float64->insertValue(value);
+    }
+    else if (auto * column_string = typeid_cast<ColumnString *>(column.get()))
+    {
+        size_t size = sizeForStrings;
+        String tmp;
+        if (header["descr"][1] == 'U')
+            size = sizeForStrings * 4;
+
+        tmp.resize(size);
+        in->readStrict(tmp.data(), size);
+        tmp.erase(std::remove(tmp.begin(), tmp.end(), '\0'), tmp.end());
+        column_string->insertData(tmp.c_str(), tmp.size());
+    }
+    else
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Error while reading data");
+}
+
+bool NpyRowInputFormat::readRow(MutableColumns & columns, RowReadExtension &  /*ext*/)
+{
+    if (in->eof())
+        return false;
+
+    if (unlikely(*in->position() == '\n'))
+    {
+        /// An empty string. It is permissible, but it is unclear why.
+        ++in->position();
+    }
```

Comment:
How is it possible? Do you have an example when we can have `\n` in the middle of the data?